README.md

```diff
@@ -1,13 +1,238 @@
+    public BigDecimal getCurrentBalance() throws BalanceException {
+        
+        //get balance from the wallet
+       
+       double balance = ...getBalance();
+       
+       //if fails to get balance throw BalanceException
+         throw new BalanceException("error details");
+         //else return the balance
+       return new BigDecimal(balance);
+    }
+}
+```
+
+These methods are asynchronous and can perform long operations
+
+
+
+## Receive Kin
+In order for an app to be able to receive Kin from other apps
+
+- In your root project add new package named *kindiscover*
+- In the kindiscover directory - create an activity class *AccountInfoActivity* and declare it in *manifest.xml* file
+
+
+```xml
+<application>
+    ...
+    <activity android:name=".kindiscover.AccountInfoActivity"
+              android:exported="true"/>       
+</application>
+```
+Very important to define the activity to be exported
+
+The *AccountInfoActivity* class must extends the abstract activity *AccountInfoActivityBase*
+and implement the abstract method:
+- *getPublicAddress* 
+
+
+```java
+public class AccountInfoActivity extends AccountInfoActivityBase {
+
+    @Override
+    public String getPublicAddress() {
+        // return the user a public address from the wallet
+        return "some address";
+    }
+    
+}
+```
+This method is asynchronous and can perform long operations
+
+
+## Receive other apps transfer notifications 
+In order to get notified about other applications that try to send kin to your app
+
+
+- In your root project add new package named *Kindiscover*
+- In the kindiscover directory - create a service class named *ReceiveKinService*, declare it in *manifest.xml* file
+and configure the service to be exported
+
+```xml
+<application>
+    ...
+    <service android:name=".kindiscover.ReceiveKinService"
+             android:exported="true" />
+</application>
+```
+
+**Notice!** Very important to define the service to be **exported**
+
+
+
+The *ReceiveKinService* class must extends the abstract service *ReceiveKinServiceBase*
+and implement the two abstract methods:
+- *onTransactionCompleted* 
+- *onTransactionFailed* 
+
+
+```java
+public class ReceiveKinService extends ReceiveKinServiceBase {
+
+    public ReceiveKinService() {
+        super();
+    }
+
+    @Override
+    public void onTransactionCompleted(@NonNull String fromAddress, @NonNull String senderAppName, @NonNull String toAddress, int amount, @NonNull String transactionId, @NonNull String memo) {
+        //This will be called when any app complete transfer Kin to your app
+    }
+
+    @Override
+    public void onTransactionFailed(@NonNull String error, @NonNull String fromAddress, @NonNull String senderAppName, @NonNull String toAddress, int amount, @NonNull String memo) {
+        //This will be called when any app failed to transfer Kin to your app
+    }
+    
+}
+```
+
+
+These methods are asynchronous and can perform long operations inside them
```

Comment:
Not true! These methods are called on the UI thread. 
If you wish to perform network or long-running operations you need to start your own background thread. The service will keep itself alive for 10 seconds after the method is called.